zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -728,6 +736,29 @@ private ManagedInterpreterGroup createInterpreterGroup(String groupId) {
     return interpreterGroup;
   }
 
+  /**
+   * Throw exception when interpreter process has already launched
+   *
+   * @param interpreterGroupId
+   * @param properties
+   * @throws IOException
+   */
+  public void setInterpreterGroupProperties(String interpreterGroupId, Properties properties)
+      throws IOException {
+    ManagedInterpreterGroup interpreterGroup = this.interpreterGroups.get(interpreterGroupId);
+    for (List<Interpreter> session : interpreterGroup.sessions.values()) {
+      for (Interpreter intp : session) {
+        if (!intp.getProperties().equals(properties) &&
+            interpreterGroup.getRemoteInterpreterProcess() != null &&
+            interpreterGroup.getRemoteInterpreterProcess().isRunning()) {
+          throw new IOException("Can not change interpreter properties when interpreter process " +
```

Comment:
I'm a bit worry about this? I think this is the way why `%dep` is so hard to use - the implicit order of interpreter is not very visible to an user, especially with different interpreter isolation mode etc.

Reply:
Right, this is what I mention in the documentation. This generic `ConfInterpreter` is for advanced users that they needs to understand the interpreter mode setting of zeppelin and be aware when interpreter process is launched.